### ISSUE TITLE
[FEA] Expose KMeans cluster initialisation to Python

### ISSUE THREAD

**Issue by betatim** [rapidsai/raft#1166]
**Describe the solution you'd like**
Expose kmeans++ cluster initialisation in pylibraft so that it can be called from Python. Maybe there is no need to have a random cluster init, `cupy.random.permutation()` might be enough.

The use case is to call the raft version of kmeans++ instead of using what is in scikit-learn. This means the arguments passed to https://github.com/scikit-learn/scikit-learn/blob/9b537396488a70de06e00de2330c4f00d7db1fc1/sklearn/cluster/_kmeans.py#L72 are available when calling the init function/having a similar function signature would be convenient.

**Additional context**
Recently there was a discussion about "k-means++", "greedy k-means++" and what is actually implemented (and what not) in https://github.com/scikit-learn/scikit-learn/issues/24973. Does the raft version match/can be configured to match "greedy k-means++"?

WDYT?

**Comment by cjnolet** [comment]
@betatim yes absolutely I think we can (and should) expose this to Python. I think this will essentially entail wrapping [this function](https://github.com/rapidsai/raft/blob/0076101e69c03bab03c9cb022d2e4c519bce60af/cpp/include/raft/cluster/kmeans.cuh#L528).

**Comment by betatim** [comment]
I'll attempt it tomorrow and see how it goes. Does that work for you? Or is there someone who can "hammer it out in 10minutes", in which case it might be more efficient to let them do it :D
